clang/lib/3C/Duplicate.cpp

```diff
@@ -0,0 +1,80 @@
+#include "clang/3C/Duplicate.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/Decl.h"
+#include "clang/3C/ProgramInfo.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+
+using namespace clang;
+using namespace llvm;
+using namespace std;
+
+
+// Recursive Visitor
+class FindVarDecl : public RecursiveASTVisitor<FindVarDecl> {
+public:
+  explicit FindVarDecl(ASTContext &Context, Rewriter &R,
+                       ProgramInfo &I, selector P)
+      : Context(Context),  R(R), Info(I), P(P) {}
+
+
+  bool VisitVarDecl(VarDecl *VD) {
+    // If the selctor matches execute the duplicate function
+    // and store the new name in the optional
+    if(P(VD)){
```

Comment:
What happens if this is true twice? I imagine you expect it won't be, but is there a reason it couldn't?

Reply:
The `return false` at the bottom should prevent this. It will stop traversing once it hits a positive